### PR TITLE
Drop in-tree qemu-img

### PR DESCRIPTION
**Pull request by aparcar**
Instead of maintaining qemu-img we can require it on the host, for the cases where VDI/VMDK images are actually required.

This doesn't fully work yet, enabling the VDI images doesn't trigger a recheck of requirements. Any advice?

### CODE REVIEW

File: include/prereq-build.mk [hunk 1]
```diff
@@ -11,6 +11,9 @@ include $(INCLUDE_DIR)/prereq.mk
 SHELL:=sh
 PKG_NAME:=Build dependency
 
+PKG_CONFIG_DEPENDS += \
```
Comment (neheb) [inline include/prereq-build.mk:14]: This only makes sense for x86 I think. No idea if it can be made conditional.

**Comment by aparcar** [comment]
Acked by @hauke via IRC

**Comment by bobafetthotmail** [comment]
as for the recheck of requirements, I think you can work around it by adding a line to the script that is actually using the tool to print a more clear error message if the qemu-img tool is not found. Like "qemu-img not found, please install it in the host".

So that the first time it may fail but the user will know how to deal with it if he checks the log or output

File: include/image-commands.mk [hunk 2]
```diff
@@ -291,8 +291,12 @@ endef
 # Convert a raw image into a $1 type image.
 # E.g. | qemu-image vdi
 define Build/qemu-image
-	qemu-img convert -f raw -O $1 $@ $@.new
-	@mv $@.new $@
+	if command -v qemu-img; then \
+		qemu-img convert -f raw -O $1 $@ $@.new; \
+		@mv $@.new $@; \
+	else \
+		echo "WARNING: Install qemu-img to create VDI/VMDK images" >&2; \
```
Comment (aparcar) [inline include/image-commands.mk:298]: Not sure if this should fail the build or just print the warning?

**Comment by oldium** [comment]
I just wanted to add support for VHDX on x86, so it is great that I do not need to provide an updated qemu-img. Thanks 👋

File: include/image-commands.mk (hunk 2 above)
Comment (dangowrt) [reply include/image-commands.mk:298]: I'd say fail, if the Build/qemu-image is called the user requested that, so it should not just fail silently and leave them wondering why.

**Comment by aparcar** [comment]
Merged.

**Pull request closed without merging**